File: exams/frmSubjectRank.vb
Public Class frmSubjectRank

    Private Sub frmSubjectRank_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            radNone.Checked = True
            radFirst.Checked = False
            radLast.Checked = False
            txtNumber.Enabled = False
            cboSubject.Items.Add(None)
            For k As Integer = 0 To subjabb.Length - 1
                cboSubject.Items.Add(subjabb(k))
            Next
            cboSubject.SelectedItem = None
        End If
    End Sub

    Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
        rank = False
        Me.Close()
    End Sub

    Private Sub btnAnalyze_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAnalyze.Click
        If cboSubject.SelectedItem <> None Then
            If (radFirst.Checked Or radLast.Checked) And Not IsNumeric(txtNumber.Text) Then
                failure("failure Value For Student Filter Number!")
                Exit Sub
            End If
            rank = True
            radF = radFirst.Checked
            radL = radLast.Checked
            Try
                rankno = txtNumber.Text
            Catch ex As Exception

            End Try
            subject = cboSubject.SelectedItem
            Me.Close()
        Else
            failure("Please Select A Subject To Analyze!")
        End If
    End Sub

    Private Sub radFirst_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles radFirst.CheckedChanged
        txtNumber.Enabled = True
    End Sub

    Private Sub radLast_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles radLast.CheckedChanged
        txtNumber.Enabled = True
    End Sub
End Class

File: exams/frmSubjectRank.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmSubjectRank
    {
        public frmSubjectRank()
        {
            InitializeComponent();
            _radLast.Name = "radLast";
            _radFirst.Name = "radFirst";
            _btnCancel.Name = "btnCancel";
            _btnAnalyze.Name = "btnAnalyze";
        }

        private void frmSubjectRank_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                radNone.Checked = true;
                radFirst.Checked = false;
                radLast.Checked = false;
                txtNumber.Enabled = false;
                cboSubject.Items.Add(publicSubsNFunctions.None);
                for (int k = 0, loopTo = publicSubsNFunctions.subjabb.Length - 1; k <= loopTo; k++)
                    cboSubject.Items.Add(publicSubsNFunctions.subjabb[k]);
                cboSubject.SelectedItem = publicSubsNFunctions.None;
            }
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            publicSubsNFunctions.rank = false;
            Close();
        }

        private void btnAnalyze_Click(object sender, EventArgs e)
        {
            if (Conversions.ToBoolean(Operators.ConditionalCompareObjectNotEqual(cboSubject.SelectedItem, publicSubsNFunctions.None, false)))
            {
                if ((radFirst.Checked | radLast.Checked) & !Information.IsNumeric(txtNumber.Text))
                {
                    publicSubsNFunctions.failure("failure Value For Student Filter Number!");
                    return;
                }

                publicSubsNFunctions.rank = true;
                publicSubsNFunctions.radF = radFirst.Checked;
                publicSubsNFunctions.radL = radLast.Checked;
                try
                {
                    publicSubsNFunctions.rankno = Conversions.ToInteger(txtNumber.Text);
                }
                catch (Exception ex)
                {
                }

                publicSubsNFunctions.subject = Conversions.ToString(cboSubject.SelectedItem);
                Close();
            }
            else
            {
                publicSubsNFunctions.failure("Please Select A Subject To Analyze!");
            }
        }

        private void radFirst_CheckedChanged(object sender, EventArgs e)
        {
            txtNumber.Enabled = true;
        }

        private void radLast_CheckedChanged(object sender, EventArgs e)
        {
            txtNumber.Enabled = true;
        }
    }
}